File: selfstudy/golang/temperatures_sortasc.go
/*
task 249:
Temperatures - SortAsc
You need to sort the values in ascending order for daily temperatures in Celsius
*/

package main

import (
	"fmt"
	"reflect"
)

func Sort_upTemperatures(temps []int) []int {
	res := make([]int, len(temps))
	copy(res, temps)
	for i := 0; i < len(res); i++ {
		for j := i+1; j < len(res); j++ {
			if res[i] > res[j] { res[i], res[j] = res[j], res[i] }
		}
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Sort_upTemperatures([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Sort_upTemperatures([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Sort_upTemperatures([]int{1, 2, 3})},
		{[]int{100, 200}, Sort_upTemperatures([]int{100, 200})},
		{[]int{}, Sort_upTemperatures([]int{})},
	}
	
	for idx, t := range tests {
		res := Sort_upTemperatures(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/temperatures_sortasc.js
/*
task 49:
Temperatures - SortAsc
You need to sort the values in ascending order for daily temperatures in Celsius
*/

function sort_up_temperatures(temps) {
  return [...temps].sort((a, b) => a - b);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: sort_up_temperatures([10, 20, 30]) },
  { input: [50, 60, 70], expected: sort_up_temperatures([50, 60, 70]) },
  { input: [1, 2, 3], expected: sort_up_temperatures([1, 2, 3]) },
  { input: [100, 200], expected: sort_up_temperatures([100, 200]) },
  { input: [], expected: sort_up_temperatures([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sort_up_temperatures', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sort_up_temperatures(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sort_up_temperatures(t.input) === t.expected || JSON.stringify(sort_up_temperatures(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
